docs/ske/05-guides/10-healthchecks.mdx

```diff
@@ -12,19 +12,116 @@ You will also need a configured and deployed Health Agent. You can do that by fo
 
 **In this tutorial you will**
 
-1. [Define a Resource Health check in a Promise](#resource-hc)
-1. [Observe the execution of the check in the destination](#observe-hc)
-1. [Observe the health check result in the requested resource](#observe-hc-result)
+1. [Generate a Health Definition in the Resource Workflow](#generate-hd)
+1. [Write a Health Check Workflow](#write-hc)
+1. [Verify the results](#observe-hc-result)
 
-## Defining a Resource Health Check {#resource-hc}
+## How do Health Checks work in SKE?
 
-In this tutorial, you will define a Resource Health Check for the Redis Promise available in the Kratix Marketplace. First, download the Promise file to your local machine:
+Health Checks are implemented in SKE via `HealthDefinition` objects. A Health Definition is the outline of the task that will be performed on a Destination to verify the health of a Resource Request. 
+
+The diagram below shows how Health Checks are processed in SKE:
+
+```mdx-code-block
+import HCDiagram from "/img/docs/ske/guide-healthcheck-arch.png"
+```
+
+<figure className="diagram">
+  <img className="large" src={HCDiagram} alt="High-level diagram of how Health Check works in Kratix" />
+
+  <figcaption>How Health Checks are processed in SKE</figcaption>
+</figure>
+
+Let's go through the diagram in detail:
+
+1. When a new resource is requested from a Promise, Kratix will do the usual and execute the Resource Configure Workflow for that Promise. 
+2. As part of the outputs of that workflow, you can include a `HealthDefinition` object, including all the necessary information to execute the Health Check Workflow on the Destination. 
+3. Once that HealthDefinition is picked up and applied at the target Destination, the Health Check agent will execute the Health Check Workflow and produce a `HealthRecord` document. 
+4. The `HealthRecord` is then written to a State Store, 
+5. The `HealthRecord` is then applied back to the Platform cluster via a GitOps agent listening to the State Store.
+6. The Health Check monitor within SKE will then process the `HealthRecord` and update the resource `status` with the results.
+
+With that in mind, let's get started!
+
+## Generate a Health Definition {#generate-hd}
+
+In this tutorial, you will define a Resource Health Check for the Redis Promise available in the Kratix Marketplace. Let's start by defining what the HealthDefinition will look like.
+
+The [HealthDefinition](/main/reference/health-checks/healthdefinition) object has the following specification:
+
+```yaml
+apiVersion: platform.kratix.io/v1alpha1
+kind: HealthDefinition
+metadata:
+  name: healthdefinition
+  namespace: default
```

Comment:
Could we please mention that Health definition can be scheduled to any namespace, not just default?